pkg/kubelet/nodestatus/setters.go

```diff
@@ -502,6 +502,9 @@ func ReadyCondition(
 		if len(missingCapacities) > 0 {
 			errs = append(errs, fmt.Errorf("missing node capacity for resources: %s", strings.Join(missingCapacities, ", ")))
 		}
+		if len(node.Status.Addresses) == 0 {
```

Comment:
Wondering if it makes more sense to set NetworkUnavailable condition instead,

Reply:
good point

Reply:
hmm, it seems that based on @danwinship excellent investigation about this condition https://github.com/kubernetes/kubernetes/issues/120486#issuecomment-1710410069 that despite the name fits, the behaviors implemented are not the ones that we want, and we should stick to the NotReady condition

Reply:
I think we probably ought to design a new "network unavailable" condition, but that's a bigger issue.